LoopKit/DeviceManager/DeviceLog/PersistentDeviceLog.swift

```diff
@@ -106,19 +108,65 @@ public class PersistentDeviceLog {
     }
 }
 
-extension NSManagedObjectContext {
+// MARK: - Simulated Core Data
+
+extension PersistentDeviceLog {
+    private var historicalEndDate: Date { Date(timeIntervalSinceNow: -.hours(24)) }
+    private var historicalEntriesPerDay: Int { 6000 }
+
+    public func generateSimulatedHistoricalDeviceLogEntries(completion: @escaping (Error?) -> Void) {
+        var startDate = Calendar.current.startOfDay(for: self.earliestLogEntryDate)
+        let endDate = Calendar.current.startOfDay(for: self.historicalEndDate)
+        var generateError: Error?
+        var entryCount = 0
+
+        self.managedObjectContext.performAndWait {
+            while startDate < endDate {
+                for index in 0..<self.historicalEntriesPerDay {
+                    let entry = DeviceLogEntry(context: self.managedObjectContext)
+                    entry.simulated(timestamp: startDate.addingTimeInterval(.hours(Double(index) * 24.0 / Double(self.historicalEntriesPerDay))))
+                    entryCount += 1
+                }
 
-    fileprivate func deleteObjects<T>(matching fetchRequest: NSFetchRequest<T>) throws -> Int where T: NSManagedObject {
-        let objects = try fetch(fetchRequest)
+                // We must save each day since there are so many historical entries
+                do {
+                    try self.managedObjectContext.save()
+                } catch let error {
+                    generateError = error
+                    return
+                }
 
-        for object in objects {
-            delete(object)
+                startDate = Calendar.current.date(byAdding: .day, value: 1, to: startDate)!
+            }
+
+            self.log.info("Generated %d historical DeviceLogEntries", entryCount)
         }
 
-        if hasChanges {
-            try save()
+        completion(generateError)
+    }
+
+    public func purgeHistoricalDeviceLogEntries(completion: @escaping (Error?) -> Void) {
+        let predicate = NSPredicate(format: "timestamp < %@", self.historicalEndDate as NSDate)
```

Comment:
I think this will only occur if the day this script is run is a DST change day. As above, this is "debug only" code for us and QA so not sure it is worth the effort to fix.